Integrated 771

Abdaaf opened the pull request:
Database v3.5

Commit:
VM added

Commit:
bundle 15:25

Commit:
Initial Commit

Commit:
Readme edited

Commit:
Readme edited

Commit:
Readme edited

Commit:
Database Sample setup

Commit:
Database v1

Commit:
Database decimal fix

Commit:
Database optional fixes

Commit:
delete entities

Commit:
Database v3.5

marijnotte commented on Tests/Controller/DefaultControllerTest.php:
```diff
@@ -0,0 +1,17 @@
+<?php
```
This seems to be an example file, which should not be commited

marijnotte commented on Controller/DefaultController.php:
```diff
@@ -0,0 +1,13 @@
+<?php
```
What does this controller do?

johanliefers replied:
Missing integrated copyright block and author tags, this counts for every document

marijnotte commented on Resources/config/routing.xml:
```diff
@@ -0,0 +1,10 @@
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/routing http://symfony.com/schema/routing/routing-1.0.xsd">
+
+    <route id="integrated_subscription_homepage" path="/hello/{name}">
```
Seems to be an example route?

marijnotte commented on Resources/views/Default/index.html.twig:
```diff
@@ -0,0 +1 @@
+Hello {{ name }}!
```
What does this do?

johanliefers commented on Entity/Subscription.php:
```diff
@@ -0,0 +1,285 @@
+/**
+ * Integrated\Bundle\SubscriptionBundle\Entity\Subscription
+ *
+ * @ORM\Entity()
```
Mapping should be in a seperate xml document. Check userbundle for example

johanliefers commented on Entity/Subscription.php:
```diff
@@ -0,0 +1,285 @@
+ * Integrated\Bundle\SubscriptionBundle\Entity\Subscription
+ *
+ * @ORM\Entity()
+ * @ORM\Table(name="subscription", indexes={@ORM\Index(name="fk_Subscription_SubscriptionType1_idx", columns={"`type`"})})
```
why make such a long index name?

johanliefers commented on Entity/Subscription.php:
```diff
@@ -0,0 +1,285 @@
+    }
+
+    /**
+     * Set the value of id.
```
unnecessary addition, you only need to add a description if there is some logic that needs explaination. In this case it is clear this function sets the value of id. This counts for all the getters and setters without any custom logic

johanliefers commented on Entity/Subscription.php:
```diff
@@ -0,0 +1,285 @@
+    /**
+     * Set SubscriptionType entity (many to one).
+     *
+     * @param \Integrated\Bundle\SubscriptionBundle\Entity\SubscriptionType $subscriptionType
```
Unnecessary to use the complete namespace

johanliefers commented on Entity/Subscription.php:
```diff
@@ -0,0 +1,285 @@
+        return $this->subscriptionType;
+    }
+
+    public function __sleep()
```
why do you add this magic function?

johanliefers commented on Entity/Subscription.php:
```diff
@@ -0,0 +1,285 @@
+     * @param \Integrated\Bundle\SubscriptionBundle\Entity\SubscriptionType $subscriptionType
+     * @return \Integrated\Bundle\SubscriptionBundle\Entity\Subscription
+     */
+    public function setSubscriptionType(SubscriptionType $subscriptionType = null)
```
subscriptiontype is not nullable, so why make it default null?

johanliefers commented on Entity/SubscriptionTypeVariant.php:
```diff
@@ -0,0 +1,228 @@
+    /**
+     * @ORM\Column(type="integer", nullable=true)
+     */
+    protected $periodlentgh;
```
use camelCase, also check other defined variables.

johanliefers commented on Entity/SubscriptionTypeVariant.php:
```diff
@@ -0,0 +1,228 @@
+     */
+    protected $subscriptionType;
+
+    public function __construct()
```
no need for constructor if it is empty

johanliefers commented on Entity/VatCountry.php:
```diff
@@ -0,0 +1,187 @@
+     */
+    public function addVatType(VatType $vatType)
+    {
+        $this->vatTypes[] = $vatType;
```
use the add function of the collection instead

johanliefers commented on LICENSE:
```diff
@@ -1,4 +1,4 @@
-Copyright (c) 2015-2016 e-Active B.V.
+Copyright (c) 2013-2014 e-Active B.V.
```
check date

johanliefers commented on README.md:
```diff
@@ -0,0 +1,49 @@
+# IntegratedBlockBundles #
+This bundle provides block management
```
Readme is of block bundle, you should create one for the subscription bundle

johanliefers commented on Resources/config/services.xml:
```diff
@@ -0,0 +1,16 @@
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+    <!--
```
Plz remove test code instead of comment it out

johanliefers commented on composer.json:
```diff
@@ -0,0 +1,37 @@
+        "symfony/framework-bundle": ">=2.2",
+        "sensio/framework-extra-bundle": "~2.3",
+        "doctrine/mongodb": ">=1.0.0",
+        "doctrine/mongodb-odm": "*",
```
why do you require odm and mongo? I think most requirements stated here are actually not really required, plz check

Commit:
Improved some things due to the review.

Commit:
Readme updated

Commit:
camelCase added in all models

Commit:
deleted all unnecessary magic function from all models

Commit:
Deleted all the full namespaces from all models

Commit:
Deleted all the unnecessary comments from all models

Commit:
Deleted all all annotation from all models

Commit:
Added all mapping documents;
Renamed most of the tablenames properly;
Foreign Key name fields deleted: doctrine assigns names automatically;

johanliefers commented on Entity/Subscription.php:
```diff
@@ -0,0 +1,211 @@
+<?php
+
```
Plz remove the auto generated comment and include the standard integrated copyright above every file

johanliefers commented on Entity/Subscription.php:
```diff
@@ -0,0 +1,211 @@
+ */
+class Subscription
+{
+    protected $id;
```
Not all annotations should be deleted, plz keep the @var tag

johanliefers commented on Entity/SubscriptionWall.php:
```diff
@@ -0,0 +1,206 @@
+
+    protected $disabled;
+
+    protected $freetier;
```
What does this variable contain (I don't know the word)?

johanliefers commented on README.md:
```diff
@@ -0,0 +1,49 @@
+# IntegratedSubscriptionWallBundles #
```
should be IntegratedSubscriptionWallBundle

johanliefers commented on README.md:
```diff
@@ -0,0 +1,49 @@
+### Import the routing ###
+
+    # app/config/routing.yml
+    integrated_block:
```
should be integrated_subscription

johanliefers commented on Resources/config/doctrine/Subscription.orm.xml:
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<doctrine-mapping xmlns="http://doctrine-project.org/schemas/orm/doctrine-mapping" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://doctrine-project.org/schemas/orm/doctrine-mapping http://doctrine-project.org/schemas/orm/doctrine-mapping.xsd">
+  <entity name="Integrated\Bundle\SubscriptionBundle\Entity\Subscription" table="subscription">
```
more white lines makes it more readable, I would suggest a new white line after every defined ORM variable

Commit:
some annotations added

johanliefers commented on DependencyInjection/IntegratedSubscriptionExtension.php:
```diff
@@ -0,0 +1,28 @@
+        $config = $this->processConfiguration($configuration, $configs);
+
+        $loader = new Loader\XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
+        $loader->load('services.xml');
```
not really necessary if you have no services yet, but you can leave it here for now, assuming that you will make some services in the future

johanliefers commented on Entity/WallChannel.php:
```diff
@@ -0,0 +1,89 @@
+
+    protected $subscriptionWall;
+
+    public function __construct()
```
remove empty constructor

johanliefers commented on composer.json:
```diff
@@ -0,0 +1,36 @@
+        "symfony/symfony": ">=2.4",
+        "symfony/framework-bundle": ">=2.2",
+        "sensio/framework-extra-bundle": "~2.3",
+        "doctrine/mongodb": ">=1.0.0",
```
why do you require odm and mongo? I think most requirements stated here are actually not really required, plz check

Fluora committed:
VAR Annotations added

Commit:
All fixes from review added

Commit:
Model or Entity fix;
author and copyrighttags addes;
composer.json: deleted some dependencies
